ganttproject/src/net/sourceforge/ganttproject/gui/ResourceAssignmentsPanel.java

```diff
@@ -0,0 +1,97 @@
+/*
+GanttProject is an opensource project management tool. License: GPL3
+Copyright (C) 2010 Dmitry Barashev
+
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU General Public License
+as published by the Free Software Foundation; either version 3
+of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package net.sourceforge.ganttproject.gui;
+
+import net.sourceforge.ganttproject.resource.HumanResource;
+import net.sourceforge.ganttproject.task.ResourceAssignment;
+import net.sourceforge.ganttproject.task.Task;
+import net.sourceforge.ganttproject.task.TaskManager;
+
+import javax.swing.*;
+import javax.swing.table.TableColumn;
+
+/**
+ * UI component in a resource properties dialog: a table with tasks assigned to
+ * a resource.
+ *
+ * @author Oleg Kushnikov
+ */
+public class ResourceAssignmentsPanel {
+  private final TaskManager myTaskManager;
+  private ResourceAssignmentsTableModel myModel;
+  private final HumanResource myPerson;
+  private JTable myTable;
+
+  public ResourceAssignmentsPanel(HumanResource person, TaskManager taskManager) {
+    myPerson = person;
+    myTaskManager = taskManager;
+  }
+
+  private JTable getTable() {
+    return myTable;
+  }
+
+  public JPanel getComponent() {
+    myModel = new ResourceAssignmentsTableModel(myPerson);
+    myTable = new JTable(myModel);
+    UIUtil.setupTableUI(getTable());
+    setUpTasksComboColumn(getTable().getColumnModel().getColumn(ResourceAssignmentsTableModel.Column.NAME.ordinal()), getTable());
+    AbstractTableAndActionsComponent<ResourceAssignment> tableAndActions =
```

Comment:
IDEA says this code is identical to code in TaskAllocationsPanel. It seems to be pretty abstract and only needs `delete` and `getAllValues()` which are not in the standard `TableModel`. Maybe just create `TableModelExt` and move this code into method `AbstractTableAndActionsComponent.create(JTable, TableModelExt)` ?